src/ethereum/ethEngine.js

```diff
@@ -524,53 +528,41 @@ export class EthereumEngine extends CurrencyEngine {
     this.addTransaction(this.currencyInfo.currencyCode, edgeTransaction)
   }
 
-  // curreently for Ethereum but should allow other currencies
+  /**
+   *  Fetch network fees from various providers in order of priority, stopping
+   *  and writing upon successful result.
+   */
   async checkUpdateNetworkFees() {
-    // Get the network fees from the info server
-    try {
-      const infoServer = getEdgeInfoServer()
-      const url = `${infoServer}/v1/networkFees/${this.currencyInfo.currencyCode}`
-      const jsonObj: EthereumFees = await this.ethNetwork.fetchGet(url)
-      const valid = asMaybe(asEthereumFees)(jsonObj) != null
-
-      if (valid) {
-        if (
-          JSON.stringify(this.walletLocalData.otherData.networkFees) !==
-          JSON.stringify(jsonObj)
-        ) {
-          this.walletLocalData.otherData.networkFees = jsonObj
-          this.walletLocalDataDirty = true
+    // Get base fees from 'rpcServers' and convert to our network fees format.
+    // * Supported for ETH post EIP-1559 fork only
+    if (this.currencyInfo.currencyCode === 'ETH') {
```

Comment:
Encapsulate this logic inside: updateNetworkFeesFromBaseFeePerGas
Then generalize this function to just loop over the provider.